features/support/github.js

```diff
@@ -0,0 +1,198 @@
+'use strict';
+const Github = require('github');
+const github = new Github();
+
+const MAX_CONTENT_LENGTH = 354;
+const MAX_FILENAME_LENGTH = 17;
+
+/**
+ * Creates a string of a given length with random alphanumeric characters
+ * @method randomString
+ * @param  {Number}     stringLength  Length of the string
+ * @return {String}                   A string consisting of random characters
+ */
+function randomString(stringLength) {
+    let content = '';
+    const alphanumeric = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
+
+    for (let i = 0; i < stringLength; i++) {
+        content += alphanumeric.charAt(Math.floor(Math.random() * alphanumeric.length));
+    }
+
+    return content;
+}
+
+/**
+ * Clean up repository
+ * @method cleanUpRepository
+ * @param  {String}          token      Github token
+ * @param  {String}          repoOwner  Owner of the repository
+ * @param  {String}          repoName   Name of the repository
+ * @param  {String}          branch     Name of the branch to delete
+ * @return {Promise}
+ */
+function cleanUpRepository(token, branch, repoOwner, repoName) {
+    const branchParams = {
+        user: repoOwner,
+        repo: repoName,
+        ref: `heads/${branch}`
+    };
+
+    // Github operations require
+    github.authenticate({
+        type: 'oauth',
+        token
+    });
+
+    return github.gitdata.getReference(branchParams)
+        .then(() => github.gitdata.deleteReference(branchParams), () => {});
+}
+
+/**
+ * Close a pull request for a given repository
+ * @method closePullRequest
+ * @param  {String}     token              Github token
+ * @param  {String}     repoOwner          Owner of the repository
+ * @param  {String}     repoName           Name of the repository
+ * @param  {Number}     prNumber           Number of the pull request
+ * @return {Promise}
+ */
+function closePullRequest(token, repoOwner, repoName, prNumber) {
+    github.authenticate({
+        type: 'oauth',
+        token
+    });
+
+    return github.pullRequests.update({
+        user: repoOwner,
+        repo: repoName,
+        number: prNumber,
+        state: 'closed'
+    });
+}
+
+/**
+ * Create a branch on the given repository
+ * @method createBranch
+ * @param  {String}     token              Github token
+ * @param  {String}     branch             Name of the branch to create
+ * @param  {String}     [repoOwner]        Owner of the repository
```

Comment:
I mean, why do you make them optional here but required in `createFile`: https://github.com/screwdriver-cd/screwdriver/blob/575fd0205959a58b7aaeb040eda6ea54e2f72f3d/features/support/github.js#L116-L117

They are very similar functions so should keep them consistent.